draft-specifications/Locators.md

```diff
@@ -0,0 +1,137 @@
+# Warning: This is a draft specification likely to undergo significant change
+
+This draft specification has been initially written based on Spritely Goblin's
+implementation of as a base for the [OCapN pre-standardization
+group](https://ocapn.org) to work from. Over time this document will change,
+likely significantly as the group  converges on the design of OCapN. If you're
+interested in being part of that  work, please join!
+
+Authors: Jessica Tallon, Christine Lemmer-Webber & The OCapN Pre-standardization
+Group.
+
+# [Introduction](#introduction)
+
+OCapN Locators used to identify OCapN capable nodes or objects available on a
+specific node. They can be used in band as Syrup encoded data or out of band
+when bootstrapping a connection as URIs.
+
+These locators are agnostic to the netlayer that the node or object is located
+on, it encodes the transport  protocol name, key and other additional data which
+would be used by any given  netlayer to reach the node.
+
+# Additional Documents
+
+This document does not stand alone, it relies upon multiple other documents
+which together build up OCapN  (Object Capability Network) specifications.
+
+This specification uses the following other specifications:
+
+-   [Syrup](): The serialization format used to encode the in band
+    representation of the locator. 
+-   [OCapN Netlayers](): Specification to open a secure communication channel
+    between two sessions, often on  different networks.
+-   [RFC3986: Uniform Resource Identifier (URI): Generic
+    Syntax](https://www.rfc-editor.org/rfc/rfc3986): For encoding URI
+    representations of locations.
+
+# [Node Locator](#ocapn-node)
+
+This identifies an OCapN node, not a specific object. This includes enough
+information to specify which netlayer and provide that netlayer with all of the
+information needed to create a  bidirectional channel to that node.
+
+The node locator include the following pieces of information (more details
+below):
+
+- **Designator**: Usually representing the key, however can be any value
+  determined by the netlayer
+- **Transport**: A unique identifier to specify a netlayer
+- **Hints**: A hashmap of additional connection information.
+
+When comparing two node locators, the designator and transport are the only
+pieces of information which need to match. Two node locators can have the same
+designator and transport but  different hints and be considered to be the same
+node.
+
+## [Syrup Serialization](#node-syrup-serialization)
+
+It's encoded as a record with the label `ocapn-node` and three arguments:
+
+```
+<ocapn-node designator  ; string
+            transport   ; symbol (cannot contain ".")
+            hints>      ; hashmap | false
```

Comment:
Would it be possible to use an empty map here, without union-ing it a subset of booleans?

Otherwise, when an empty map is provided, in which sense it is distinct from `false`?

Reply:
Yeah, I think it could be an empty map, I'm not sure which is better. I'm certainly open to the change though.

It's usage of `false` here is mostly due to its roots in scheme which uses false as a value when something is lacking a specific value, like a lot of languages use a void/null like type (syrup doesn't have a other null like type).